integration_test/cases/repo.exs

```diff
@@ -881,6 +881,39 @@ defmodule Ecto.Integration.RepoTest do
             %Comment{text: "6", lock_version: 6}] = inserted_rows
   end
 
+  @tag :insert_select
+  test "insert_all with source query" do
+    adapter = System.get_env("ECTO_ADAPTER")
+
+    {:ok, %Post{id: id}} = TestRepo.insert(%Post{
+      title: "A generic title"
+    })
+
+    source = from p in Post,
+      select: %{
+        title: fragment("concat(?, ?, ?)", p.title, type(^" suffix ", :string), p.id)
+      }
+
+    opts = case adapter do
+      "pg" -> [conflict_target: [:id], on_conflict: :replace_all, returning: [:id, :title]]
+      "myxql" -> [on_conflict: :replace_all]
+      "tds" -> [returning: [:id, :title]]
+    end
```

Comment:
Sorry, I probably gave unclear advice. Those should be distinct tests:

1. one test for insert_all with query and conflict target
2. one test for insert_all with query and returning
3. one test for insert_all with query and on_conflict
4. one test for insert_all with query

Then mark each test with its appropriate tags so they can be skipped from ecto_sql. :) We should not depend on the adapter here.

Reply:
Feel free to put all of them in a shared describe block called "insert_all with query". :)

Reply:
I see. I think the latest push is how you wanted it to work.